core/src/main/scala/org/http4s/parser/AcceptCharsetHeader.scala

```diff
@@ -11,35 +11,38 @@
 package org.http4s
 package parser
 
+import cats.parse.Parser1
 import cats.syntax.all._
-import org.http4s.internal.parboiled2._
 import org.http4s.CharsetRange._
-import org.http4s.QValue.QValueParser
+import org.http4s.headers.`Accept-Charset`
 
 private[parser] trait AcceptCharsetHeader {
-  def ACCEPT_CHARSET(value: String): ParseResult[headers.`Accept-Charset`] =
-    new AcceptCharsetParser(value).parse
-
-  private class AcceptCharsetParser(input: ParserInput)
-      extends Http4sHeaderParser[headers.`Accept-Charset`](input)
-      with QValueParser {
-    def entry: Rule1[headers.`Accept-Charset`] =
-      rule {
-        oneOrMore(CharsetRangeDecl).separatedBy(ListSep) ~ EOL ~> { (xs: Seq[CharsetRange]) =>
-          headers.`Accept-Charset`(xs.head, xs.tail: _*)
+  def ACCEPT_CHARSET(value: String): ParseResult[`Accept-Charset`] =
```

Comment:
I've been putting the cats-parse parsers on the companion object of whatever is being parsed.  This way, it has access to private constructors to only accept valid values.  But this works for now.

Reply:
Done